src/axl.h

```diff
@@ -45,26 +45,59 @@ typedef enum {
     AXL_XFER_PTHREAD,      /* parallel copy using pthreads */
 } axl_xfer_t;
 
-/** Read configuration from non-AXL-specific file
- * Also, start up vendor specific services */
-int AXL_Init (const char* state_file);
+#define ARG0(dummy, a0, ...) a0
+#define GET_ARG0(...) ARG0(dummy, ## __VA_ARGS__, 0)
+
+/*
+ * int AXL_Init(void) - Initialize the library.
+ *
+ * NOTE: AXL_Init() used to take in a state_file argument, but this has
+ * since been removed. We do some macro mangling to make sure both state_file
+ * version and the no-arg version both work, but the state_file version is
+ * deprecated and should not be used.
+ */
+#define AXL_Init(...) __AXL_Init(GET_ARG0(__VA_ARGS__))
+int __AXL_Init (const char* state_file);
 
 /** Shutdown any vendor services */
 int AXL_Finalize (void);
 
-/** Create a transfer handle (used for 0+ files)
- * Type specifies a particular method to use
- * Name is a user/application provided string
- * Returns an ID to the transfer handle,
- * Returns -1 on error */
-int AXL_Create (axl_xfer_t type, const char* name);
+/*
+ * int AXL_Create(type, name, state_file) - Create a transfer handle to copy files
+ *
+ * type:        Transfer type to use
+ * name:        A unique tag to give to this transfer handle
+ * state_file:  (optional) Path to store our transfer state.  This is needed to
+ *              resume transfers after a crash.
+ *
+ * Returns an AXL ID, or negative number on error.
+ *
+ * NOTE: AXL_Create() used to only take in xtype and name.  The state_file arg
+ * was added later.  We do some macro mangling to make sure both 2-arg and
+ * 3-arg versions of AXL_Create() work, but the 2-arg version is deprecated
+ * and should not be used.
+ */
+#define AXL_Create(type, name, ...) \
+        __AXL_Create(type, name, GET_ARG0(__VA_ARGS__))
+int __AXL_Create(axl_xfer_t xtype, const char* name, const char *state_file);
```

Comment:
We need to define whether it's an error if someone names a state file, which exists and is read in, but then they have provide an xtype and/or name here that do not match the values recorded in the file.